engine/api/src/main/scala/pl/touk/nussknacker/engine/api/typed/DynamicGlobalVariable.scala

```diff
@@ -0,0 +1,13 @@
+package pl.touk.nussknacker.engine.api.typed
+
+import pl.touk.nussknacker.engine.api.MetaData
+import pl.touk.nussknacker.engine.api.typed.typing.TypingResult
+
+/**
+  * Trait to be implemented by global variable to provide dynamic value and more detailed typing info.
+  */
+trait DynamicGlobalVariable {
+  def value(metadata: MetaData): Any
```

Comment:
maybe one method returning sth like ObjectWithType would be better?

Reply:
It could cause accidential put to much logic to types computation step. IMO current approach is better.